src/cairoUtilFuncGen/calldata/calldataToMemory.ts

```diff
@@ -96,44 +96,40 @@ export class CallDataToMemoryGen extends StringIndexedFuncGen {
     if (isReferenceType(elementT)) {
       const recursiveFunc = this.getOrCreateFuncDef(elementT);
       copyCode = endent`
-        let cdElem = calldata[0];
-        let (mElem) = ${recursiveFunc.name}(cdElem);
-        dict_write{dict_ptr=warp_memory}(mem_start, mElem);
+        let calldata_elem = calldata[index];
+        let memory_elem = ${recursiveFunc.name}(calldata_elem);
+        warp_memory.unsafe_write(calldata_elem, memory_elem)
       `;
       auxFunc = recursiveFunc;
     } else if (memoryElementWidth === 2) {
       copyCode = endent`
-        dict_write{dict_ptr=warp_memory}(mem_start, calldata[0].low);
-        dict_write{dict_ptr=warp_memory}(mem_start+1, calldata[0].high);
+        warp_memory.unsafe_write(mem_ptr, calldata[index].low)
+        warp_memory.unsafe_write(mem_ptr + 1, calldata[index].high)
       `;
-      auxFunc = this.requireImport(...DICT_WRITE);
+      auxFunc = this.requireImport(...WM_UNSAFE_WRITE);
     } else {
-      copyCode = `dict_write{dict_ptr=warp_memory}(mem_start, calldata[0]);`;
-      auxFunc = this.requireImport(...DICT_WRITE);
+      copyCode = `warp_memory.unsafe_write(mem_start, calldata[index]);`;
+      auxFunc = this.requireImport(...WM_UNSAFE_WRITE);
     }
 
     const funcName = `cd_to_memory_dynamic_array${this.generatedFunctionsDef.size}`;
     return {
       name: funcName,
       code: endent`
-        #[implicit(warp_memory)]
-        func ${funcName}_elem(calldata: ${callDataType.vPtr}, mem_start: felt, length: felt){
-            alloc_locals;
-            if (length == 0){
+        #[implicit(warp_memory: WarpMemory)]
+        fn ${funcName}_elem(calldata: ${callDataType}, mem_ptr: felt252, length: felt252) {
+            if index == length {
                 return ();
             }
             ${copyCode}
-            return ${funcName}_elem(calldata + ${
-        callDataType.vPtr.to.width
-      }, mem_start + ${memoryElementWidth}, length - 1);
+            ${funcName}_elem(calldata, mem_ptr + ${memoryElementWidth}, index + 1, length)
         }
-        #[implicit(warp_memory)]
-        func ${funcName}(calldata : ${callDataType}) -> (mem_loc: felt){
-            alloc_locals;
-            let (len256) = felt_to_uint256(calldata.len);
-            let (mem_start) = wm_new(len256, ${uint256(memoryElementWidth)});
-            ${funcName}_elem(calldata.ptr, mem_start + 2, calldata.len);
-            return (mem_start,);
+
+        #[implicit(warp_memory: WarpMemory)]
+        fn ${funcName}(calldata : ${callDataType}) -> felt{
+            let mem_start = warp_memory.new_dynamic_array(calldata.len, ${memoryElementWidth});
+            ${funcName}_elem(calldata, mem_start + 1, calldata.len);
```

Comment:
${funcName}_copy or _fill imo would be a better name

Reply:
My only issue against this is that all copy semantics have the same name. I would postpone it because to do it the right way we should update the name of every copy functions